commons-numbers-complex-arrays/src/main/java/org/apache/commons/numbers/complex/arrays/ComplexList.java

```diff
@@ -37,6 +37,17 @@
  */
 public abstract class ComplexList extends AbstractList<Complex> {
 
+    /**
+     * The maximum size of array to allocate.
+     * Ensuring max capacity is even with additional space for VM array headers.
+     */
+    protected static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 9;
```

Comment:
These should not be here. The max capacity of the interleaved list is different from the non-interleaved list. Only the DEFAULT_CAPACITY should be shared.